test/test_utils/security.py

```diff
@@ -1092,7 +1066,289 @@ def fetch_other_vulnerability_lists(image, ecr_client, minimum_sev_threshold):
+            "upgradable_packages_data_for_impacted_packages", {}
+        ):
+            package_related_binaries = sorted(
+                embedded_apt_patch_evaluation_data[
+                    "upgradable_packages_data_for_impacted_packages"
+                ][package]
+            )
+            ignore_message = f"""{ignore_message} Packages: {",".join(package_related_binaries)} have been upgraded."""
+    return is_package_upgradable, ignore_message
+
+
+def check_if_python_vulnerability_is_non_patchable_and_get_ignore_message(
+    vulnerability: AllowListFormatVulnerabilityForEnhancedScan,
+    package_version_dict: dict,
+    docker_exec_command: str,
+):
+    """
+    This method takes in the vulnerability Object and checks if it is non-patchable (or allowlistable in other words). It makes the following checks:
+    1. Check if the installed package version is same as the one being shown by scanner, if not declare it as non-patchable/allowlistable
+    2. Check if the package is already in the latest version, if not declare it as non-patchable/allowlistable
+
+    :param vulnerability: Object of type AllowListFormatVulnerabilityForEnhancedScan, vuln that we need to check patchability for.
+    :param package_version_dict: Dict, key = package name and value = package version
+    :param docker_exec_command: str, Docker exec command to run additional commands on the container
+    :return: [bool, str], returns 2 values, the first says True if the vulnerability/package is non-patchable and the second one stores the ignore message
+             in case the vulnerability is non-patchable. This ignore message is used to insert into the allowlist.
+    """
+    assert (
+        vulnerability.package_details.package_manager == "PYTHONPKG"
+    ), f"Vulnerability: {json.dumps(vulnerability, cls=EnhancedJSONEncoder)} is not PythonPkg managed."
+
+    package_name = vulnerability.package_name.lower()
+    if package_name not in package_version_dict:
+        # To begin with, we will not allowlist package names that are not present in the installed list and show up in the vulnerability.
+        # However based on the observed behavior, or such scenarios occuring in the future, we will start to allowlist this vulnerability.
+        LOGGER.info(f"Package {package_name} not found!")
+        return False, ""
+
+    ## 1. Check if the installed package version is same as the one being shown by scanner
+    installed_package_version = package_version_dict.get(package_name)
+    vulnerability_package_version = vulnerability.package_details.version
+    if installed_package_version != vulnerability_package_version:
+        return (
+            True,
+            f"Installed package version is {installed_package_version} which is not equal to the one shown in vulnerability.",
+        )
+
+    ## 2. Check if the package is already in the latest version, if not then allowlist
+    response = requests.get(f"https://pypi.org/pypi/{package_name}/json")
```

Comment:
Put in a helper function that uses `retry` from a pre-existing library  - `tenacity`

Reply:
Done